specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,959 @@
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// Options for unpacking a package
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass UnbundleOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into UnpackPackage
+        /// </summary>
+        UnbundleOptions();
+
+        /// <summary>
+        /// The output folder to unpack the bundle package into.
+        /// </summary>
+        String UnbundledPackageRootDirectory{ get; set; };
+        /// <summary>
+        /// If true, the operation will overwrite existing files in the UnbundledPackageRootDirectory
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// 
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass CreateMountableImageOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into CreateMountableImage
+        /// </summary>
+        CreateMountableImageOptions();
+
+        /// <summary>
+        /// Output path for the image file.
+        /// Valid file extensions are vhd, vhdx, and CIM.
+        /// </summary>
+        String ImageFilePath{ get; set; };
+
+        /// <summary>
+        /// Root directory path for packages inside the image.
+        /// Defaults to "WindowsApps"
+        /// </summary>
+        String PackageRootDirectoryInImage{ get; set; };
+
+        /// <summary>
+        /// If true, the operation will overwrite an existing file in the ImageFilePath
+        /// when unpacking.
+        /// If false, the operation will fail if the file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// Static methods for creating and unpacking packages.
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass MakeMSIXManager
+    {
+        /// <summary>
+        /// Creates packages from the directoryPathToPack as specified by the packOptions.
+        /// </summary>
+        static Windows.Foundation.IAsyncAction Pack(String directoryPathToPack, PackOptions packOptions);
+        /// <summary>
+        /// Creates bundles from the directoryPathToBundle as specified by the bundleOptions.
+        /// </summary>
+        static Windows.Foundation.IAsyncAction Bundle(String directoryPathToBundle, BundleOptions bundleOptions);
+        /// <summary>
+        /// Unpacks a package at packageFilePathToUnpack as specified by the unpackOptions.
+        /// </summary>
+        static Windows.Foundation.IAsyncAction Unpack(String packageFilePathToUnpack, UnpackOptions unpackOptions);
```

Comment:
SUGGESTION: rename `packageFilePathToUnpack` to `inputFileName`

These functions' 1st parameter tries to convey 2 pieces of information

1. This is the input or source file
2. This is a package/bundle

Which leads to somewhat complex and confusing naming.

SUGGESTION: Drop #2 and just name the input parameter something simple and unambiguous, e.g. `inputFileName`